D3521R0.md

```diff
@@ -200,80 +200,46 @@ Given an example like:
-The pattern matches if `__temp` contextually converts to `true` and
+More precisely: Let `E` be
+`std::try_cast<std::remove_cv_t<@*type*@>>(@*subject*@)`.  The match succeeds
+if and only if: `E` is well-formed; `E` is implicitly convertible to
+`@*type*@`; `E` contextually converts to `true`; and
 `std::forward_like<decltype(@*subject*@)>(*__temp)` matches *subpattern*.
 
 # Proposed Wording
 
-### [any.synop]{.sref} Header `<any>` synopsis {.unnumbered .unlisted}
+### Insert after `[utility.unreachable]` in header `<utilities>` synopsis ([utility.syn]{.sref}):
 
 ```diff
-  template<class T>
-    T any_cast(const any& operand);
-  template<class T>
-    T any_cast(any& operand);
-  template<class T>
-    T any_cast(any&& operand);
-
-  template<class T>
-    const T* any_cast(const any* operand) noexcept;
-  template<class T>
-    T* any_cast(any* operand) noexcept;
-
-+ template<class T, same_as<any> Any>
-+   const T* try_cast(const Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any&& operand) noexcept;
++ // [utility.try_cast], `try_cast`
++ template<class T>
++   requires same_as<T, remove_cvref_t<T>>
++     constexpr @*unspecified*@ try_cast = @*unspecified*@;
 ```
 
-### [any.nonmembers]{.sref} Non-member functions {.unnumbered .unlisted}
-
-```diff
-+ template<class T, same_as<any> Any>
-+   const T* try_cast(const Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any&& operand) noexcept;
-```
+### Append to [utility]{.sref}:
 
 ::: add
-> *Mandates*: `is_void_v<T>` is `false`.
->
-> *Returns*: `any_cast<T>(&operand)`
-:::
+???.?.? `try_cast`                                   [utility.try_cast]
 
-### [exception.syn]{.sref} Header `<exception>` synopsis {.unnumbered .unlisted}
+[1]{.pnum} The name `try_cast` denotes a customization point object ([customization.point.object]).
 
-```diff
-  exception_ptr current_exception() noexcept;
-  [[noreturn]] void rethrow_exception(exception_ptr p);
-  template <class E>
-    const E* exception_ptr_cast(const exception_ptr& p) noexcept;
-  template <class T> [[noreturn]] void throw_with_nested(T&& t);
-
-+ template<class E>
-+   const E* try_cast(const exception_ptr& p) noexcept;
-```
+[2]{.pnum} Given a subexpression `E` with type `U`, let `u` be an lvalue that denotes the reified object for `E`. Then:
 
-### [propagation]{.sref} Exception propagation {.unnumbered .unlisted}
+ - [2.1]{.pnum} If `E` is an rvalue, `try_cast(E)` is ill-formed.
 
-```diff
-+ template<class E>
-+   const E* try_cast(const exception_ptr& p) noexcept;
-```
+ - [2.2]{.pnum} Otherwise, if `same_as<remove_cvref_t<U>, any>` is `true` ([any]), then `try_cast(E)` is expression-equivalent to `any_cast<T>(&u)` if that is well-formed, and is ill-formed otherwise.
```

Comment:
All the interesting requirements are _Mandates_ on `T` so this won't work. I think just say it's equivalent to `any_cast<T>(&u)` since what else can we do.